### PR TITLE
Prevent global `CallError` tracker from growing indefinitely

### PR DESCRIPTION
**Pull request by Bromeon**
Implements a ring buffer to limit the number of simultaneously available `CallError` instances. Hands out IDs and guarantees that IDs become invalid once the slot is overwritten.

### CODE REVIEW

**Comment by GodotRust** [comment]
API docs are being generated and will be shortly available at: https://godot-rust.github.io/docs/gdext/pr-798

File: godot-core/src/private.rs [hunk 1]
```diff
@@ -41,24 +40,57 @@ sys::plugin_registry!(pub __GODOT_PLUGIN_REGISTRY: ClassPlugin);
...
+    generation: u16,
+}
+
+impl Default for CallErrors {
+    fn default() -> Self {
+        Self {
+            // [None; N] requires Clone.
+            ring_buffer: std::iter::repeat_with(|| None)
+                .take(Self::MAX_ENTRIES as usize)
+                .collect(),
```
Comment (lilizoey) [inline godot-core/src/private.rs:55]: Just been going back through PRs i didnt get a chance to look at but just fyi you can do this now:
```rs
[const { None }; N]
```

Comment (Bromeon) [reply godot-core/src/private.rs:55]: Oh, this is great to know, thanks! I was looking for alternative patterns but couldn't find this, maybe too recent :slightly_smiling_face: 

Will introduce as a drive-by-commit somewhere :+1:

Comment (Bromeon) [reply godot-core/src/private.rs:55]: Unfortunately with `vec![...]` it doesn't work:

```rs
error: no rules expected the token `const`
  --> godot-core/src/private.rs:53:31
   |
53 |             ring_buffer: vec![const { None }; Self::MAX_ENTRIES as usize],
   |                               ^^^^^ no rules expected this token in macro call
   |
   = note: while trying to match end of macro
```

And directly `None` caused the original error:
```rs
error[E0277]: the trait bound `Option<call_error::CallError>: Clone` is not satisfied
    --> godot-core/src/private.rs:53:32
     |
53   |             ring_buffer: vec![ None; Self::MAX_ENTRIES as usize],
     |                          ------^^^^-----------------------------
     |                          |     |
     |                          |     the trait `Clone` is not implemented for `Option<call_error::CallError>`
     |                          required by a bound introduced by this call
     |
     = note: required for `Option<call_error::CallError>` to implement `Clone`
```

---

What does work is this, but then again that's not really better than the current code:
```rs
            ring_buffer: [const { None }; Self::MAX_ENTRIES as usize]
                .into_iter()
                .collect(),
```

Comment (lilizoey) [reply godot-core/src/private.rs:55]: can't you just do `.into()`?

Comment (Bromeon) [reply godot-core/src/private.rs:55]: I thought I tried that, but probably made a mistake. That works, thanks! :slightly_smiling_face:

**Pull request merged**